src/meta/src/hummock/compaction/picker/base_level_compaction_picker.rs

```diff
@@ -346,9 +386,20 @@ impl LevelCompactionPicker {
                     target_sub_level_id: level.sub_level_id,
                 });
             }
+        }
 
+        {
+            // report
             if skip_by_write_amp {
-                stats.skip_by_write_amp_limit += 1;
+                stats.set_skip_by_write_amp_limit();
+            }
+
+            if skip_by_count {
+                stats.set_skip_by_count_limit();
+            }
+
+            if skip_by_pending {
+                stats.set_skip_by_pending_files();
             }
```

Comment:
nits: We can call `stats.set_skip_by_xxx` directly in L303 and 308 without introducing the extra booleans.

Reply:
This is inappropriate, if we exit at line 472 skip metrics should not be set to true